### PR TITLE
Update Heroku provision to throw 409 on unique conflict

### PR DESCRIPTION
**Pull request by bechurch**
## Summary
Update the Heroku Provision endpoint to throw a 409 AlreadyExists when a unique constraint error is thrown.
image

## After Merge
* [ ] Does this change invalidate any docs or tutorials? _If so ensure the changes needed are either made or recorded_
* [ ] Does this change require a release to be made? Is so please create and deploy the release

### CODE REVIEW

**Comment by expede** [comment]
I'm actually not totally sure what Heroku expects from their docs

> Requests MAY be sent multiple times (Heroku follows the pattern of at least once delivery), and endpoints MUST respond idempotently

Idempotency in the DB, or in how we respond, as well?

**Comment by expede** [comment]
There we go:

> The add-on provision request POST /heroku/resources sent from Heroku to the partner is expected to be idempotent. In other words, the same request MAY be sent to the partner multiple times, but only a single resource SHOULD be created per uuid, and the same response SHOULD be returned each time. However, if the resource has been deprovisioned, it SHOULD NOT be created again and a 410 SHOULD be returned. In order to facilitate data expunging by the partner the time frame in which a provision request may be retried is limited to 24 hours.

So, we should respond with a `201 Created` if they send the same request again

File: library/Fission/User/Mutation.hs [hunk 1]
```diff
@@ -29,14 +29,18 @@ createWithHeroku ::
...
+        , herokuAddOnInsertedAt = now
+        , herokuAddOnModifiedAt = now
+        }
+  insertUnique herokuAddOnRecord >>= \case
+    Just addOnId ->
+      create username password Nothing (Just addOnId) now
 
-  create username password Nothing (Just addOnId) now
+    Nothing ->
+      return (Left Error.AlreadyExists)
```
Comment (expede) [inline library/Fission/User/Mutation.hs]: Oh, this is great, though! At the call site, it can capture this is a normal `Left` value and return the `201` 👍

Comment (expede) [reply library/Fission/User/Mutation.hs]: Good job on the internal error modelling 🎉

Comment (bechurch) [reply library/Fission/User/Mutation.hs]: Ok so in this case whats the best way to handle their password 😬 ? 

We can't return it. we can create a new one but then technically it isn't idempotent anymore.

Comment (expede) [reply library/Fission/User/Mutation.hs]: Yeah, good point. The docs aren't clear so I'll email them 🤷‍♀

Comment (bechurch) [reply library/Fission/User/Mutation.hs]: @expede If you do so can you cc me :)

Comment (bechurch) [reply library/Fission/User/Mutation.hs]: @expede any response?

**Comment by bechurch** [comment]
@expede This is update and ready for review.

Note that after our call yesterday we decided to leave this as a 409

File: library/Fission/User/Mutation.hs (hunk 1 above)
Comment (expede) [reply library/Fission/User/Mutation.hs]: Not yet 🤷‍♀ Whatever let's see if this causes issues

**Pull request merged**